### ISSUE TITLE
`feature.local_binary_pattern` fails on empty array sliced from non-contiguous source

### ISSUE THREAD

**Issue by soupault** [scikit-image/scikit-image#2053]
## Description

There is a bug somewhere, because `local_binary_pattern` has [np.ascontiguousarray](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/texture.py#L295) call inside.

```
Traceback (most recent call last):
  File "sss.py", line 12, in <module>
    lbp = feature.local_binary_pattern(x, 8, 1, 'uniform')
  File "/usr/lib/python3.5/site-packages/skimage/feature/texture.py", line 296, in local_binary_pattern
    output = _local_binary_pattern(image, P, R, methods[method.lower()])
  File "skimage/feature/_texture.pyx", line 84, in skimage.feature._texture._local_binary_pattern (skimage/feature/_texture.c:2505)
ValueError: Buffer not C contiguous.
```
## Way to reproduce

```
import numpy as np
from skimage import feature


z = np.random.random(200**2).reshape((200, 200))
# Works
# x = z[30:10, 10:30]
# Fails
x = z[10:30, 30:10]
print(x.shape, x.size)

lbp = feature.local_binary_pattern(x, 8, 1, 'uniform')
```
- [x] Operating system and version: `Linux 4.5.0-1-ARCH x86_64 GNU/Linux`
- [x] Python version: `Python 2.7.11, Python 3.5.1`
- [x] scikit-image version (run `skimage.__version__`): `0.12.3`

**Comment by soupault** [comment]
As a solution I can suggest adding a check for 0-long dimensions in [assert_nD](https://github.com/scikit-image/scikit-image/blob/master/skimage/_shared/utils.py#L147) and adding `assert_nD` to `feature.local_binary_pattern`.

**Comment by Bernwang** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [`feature.local_binary_pattern` fails on empty array sliced from non-contiguous source #2053](https://trello.com/c/iswzASUv/7-feature-local-binary-pattern-fails-on-empty-array-sliced-from-non-contiguous-source-2053)

**Comment by soupault** [comment]
Closed via #2364.
